packages/neon/neon/lib/src/widgets/user_avatar.dart

```diff
@@ -0,0 +1,199 @@
+// ^ This is a really strange false positive, it goes of at a very random place without any meaning. Hopefully fixed soon?
+
+part of '../../neon.dart';
+
+const kAvatarSize = 40.0;
+
+class NeonUserAvatar extends StatefulWidget {
+  const NeonUserAvatar({
+    required this.account,
+    this.username,
+    this.backgroundColor,
+    this.foregroundColor,
+    this.showStatus = true,
+    this.size = kAvatarSize,
+    super.key,
+  });
+
+  final Account account;
+  final String? username;
+  final Color? backgroundColor;
+  final Color? foregroundColor;
+  final bool showStatus;
+  final double size;
+
+  @override
+  State<NeonUserAvatar> createState() => _UserAvatarState();
+}
+
+class _UserAvatarState extends State<NeonUserAvatar> {
+  late final _userStatusBloc = Provider.of<AccountsBloc>(context, listen: false).getUserStatusesBloc(widget.account);
+
+  String get effectiveUsername => widget.username ?? widget.account.client.username!;
+
+  @override
+  void initState() {
+    super.initState();
+
+    if (widget.username != null && widget.showStatus) {
+      unawaited(_userStatusBloc.load(widget.username!));
+    }
+  }
+
+  @override
+  Widget build(final BuildContext context) {
+    final isDark = Theme.of(context).brightness == Brightness.dark;
+    final size = (widget.size * MediaQuery.of(context).devicePixelRatio).toInt();
+    return Stack(
+      alignment: Alignment.center,
+      children: [
+        CircleAvatar(
+          radius: widget.size / 2,
+          child: ClipOval(
+            child: NeonCachedApiImage(
+              account: widget.account,
+              cacheKey: 'avatar-$effectiveUsername-${isDark ? 'dark' : 'light'}$size',
+              download: () async {
+                if (isDark) {
+                  return widget.account.client.core.getDarkAvatar(
+                    userId: effectiveUsername,
+                    size: size,
+                  );
+                } else {
+                  return widget.account.client.core.getAvatar(
+                    userId: effectiveUsername,
+                    size: size,
+                  );
+                }
+              },
+            ),
+          ),
+        ),
+        CircleAvatar(
+          radius: widget.size / 2,
+          child: ClipOval(
+            child: NeonCachedApiImage(
+              account: widget.account,
+              cacheKey: 'avatar-$effectiveUsername-${isDark ? 'dark' : 'light'}$size',
+              download: () async {
+                if (isDark) {
+                  return widget.account.client.core.getDarkAvatar(
+                    userId: effectiveUsername,
+                    size: size,
+                  );
+                } else {
+                  return widget.account.client.core.getAvatar(
+                    userId: effectiveUsername,
+                    size: size,
+                  );
+                }
+              },
+            ),
+          ),
+        ),
```

Comment:
wdym? The refactoring was the addition that now the avatar and status can be fetched for any users, not only the current account.

Reply:
Oh sorry for not being clear. This is two times exactly the same code and even in a stack. 
So you are stacking the same thing on itself  and I can't see a purpose.

I thought this might have been a rebase gone rouge

Reply:
Ah I didn't look at it, but it's probably a bad rebase indeed